internal/core/hostid/azure.go

```diff
@@ -49,5 +53,19 @@ func AzureUniqueID() string {
 		return ""
 	}
 
-	return fmt.Sprintf("%s/%s/microsoft.compute/virtualmachines/%s", compute.Doc.SubscriptionID, compute.Doc.ResourceGroupName, compute.Doc.Name)
+	if compute.Doc.VMScaleSetName == "" {
+		return fmt.Sprintf("%s/%s/microsoft.compute/virtualmachines/%s", compute.Doc.SubscriptionID, compute.Doc.ResourceGroupName, compute.Doc.Name)
+	}
+
+	parsedName := strings.Split(compute.Doc.Name, "_")
+
+	// names of VM's in VMScalesets seem to follow the of `<scale-set-name>_<instance-id>`
+	// where scale-set-name is alphanumeric (and is the same as compute.vmScaleSetName
+	// field from the metadata endpoint)
+	if len(parsedName) != 2 {
+		return ""
+	}
+
+	return fmt.Sprintf("%s/%s/microsoft.compute/virtualmachinescalesets/%s/virtualmachines/%s", compute.Doc.SubscriptionID, compute.Doc.ResourceGroupName, compute.Doc.VMScaleSetName, parsedName[1])
```

Comment:
Is there logic already in our backend that also generates this for the Azure metric API integration?

Reply:
Marking as resolved as per our conversation